core/sugartypes.ml

```diff
@@ -1,42 +1,56 @@
 open CommonTypes
 open Operators
+open SourceCode
 open Utility
 
 (** The syntax tree created by the parser. *)
 
 type name = string [@@deriving show]
 
-type position = SourceCode.pos
-let dummy_position = SourceCode.dummy_pos
+module Binder : sig
```

Comment:
I really don't like having a module signature in `*.ml` file. Feels redundant. I understand this is for the sake of `traverse_map`, which has a non-trivial type. I changed this by removing the module signature and then just putting a type annotation on `traverse_map` only. Yell if you have a strong opinion on this.